azure/create-platform-prerequisites/main.tf

```diff
@@ -275,6 +275,45 @@ resource "azurerm_resource_group" "platform_rg" {
   }
 }
 
+resource "azuread_application" "babylon" {
+  display_name     = "${local.pre_name}Babylon${local.post_name}"
+  logo_image       = filebase64("cosmotech.png")
+  owners           = data.azuread_users.owners.object_ids
+  sign_in_audience = var.audience
+
+  tags = ["HideApp", "WindowsAzureActiveDirectoryIntegratedApp", var.project_stage, var.customer_name, var.project_name, "terraformed"]
+
+  required_resource_access {
+    resource_app_id = "00000003-0000-0000-c000-000000000000" # Microsoft Graph
+
+    resource_access {
+      id   = "e1fe6dd8-ba31-4d61-89e7-88639da4683d" # User.Read
+      type = "Scope"
+    }
+  }
+
+  required_resource_access {
+    resource_app_id = azuread_application.platform.application_id # Cosmo Tech Platform
+
+    resource_access {
+      id   = "6332363e-bcba-4c4a-a605-c25f23117400" # platform
+      type = "Scope"
+    }
+  }
+
+  public_client {
+    redirect_uris = ["http://localhost:8484/"]
+  }
+}
+
+resource "azuread_service_principal" "babylon" {
+  depends_on                   = [azuread_service_principal.swagger]
+  application_id               = azuread_application.babylon[0].application_id
```

Comment:
I have commented by error on commit but to quickly say the same thing babylon[0] should be used only if there is a "Count=..." on the app registration. Add a var to make the app registration optional or change the application_id to azuread_application.babylon.application_id.
A "terraform validate" command may see this kind of issue.